src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleThreadContext.java

```diff
@@ -171,6 +191,23 @@ void addImport(String localRepoName, String exportedName, String byWhat, Locatio
       imports.put(localRepoName, exportedName);
     }
 
+    public void addRepoOverride(
+        String exportedName,
+        String localRepoName,
+        ImmutableList<StarlarkThread.CallStackEntry> stack)
+        throws EvalException {
+      RepositoryName.validateUserProvidedRepoName(exportedName);
+      RepositoryName.validateUserProvidedRepoName(localRepoName);
+      if (repoOverrides.containsKey(exportedName)) {
```

Comment:
optional -- this `if` can be merged with the follow `put` (see `addRepoNameUsage` above):

```java
RepoOverride collision = repoOverrides.put(...);
if (collision != null) { ... }
```